src/jit/importer.cpp

```diff
@@ -7522,7 +7522,9 @@ var_types Compiler::impImportCall(OPCODE                  opcode,
             }
         }
 
-        if (!bIntrinsicImported)
+        // Don't create cast for a variable on the stack if it is intrinsic or tail call.
+        // We assume that the value will not be physically represented there.
+        if (!bIntrinsicImported && !(tailCall && canTailCall))
```

Comment:
Yes, if we set checkSmallType = true and forbid tail calls in morph, we will get movzx before the commit and mov after this change. It is wrong.
To fix it we can:
1) Move checkForSmallType logic from importer to morph;
2) Create the cast during importation and delete it in morph;
What is better?